### ISSUE TITLE
FakeQuerySets can't delete

### ISSUE THREAD

**Issue by johannes-torchbox** [torchbox/django-modelcluster#14]
I'm wondering about a strange situation I'm in. I'm doing something like this:

       model_name.related_name.all().delete()

And I get:

    AttributeError: 'FakeQuerySet' object has no attribute 'delete'

There are related items and they are also *not* shown by the FakeQuerySet, so there's another problem there. This only happens in web requests, not on the console, and I have no idea where the FakeQuerySet comes from.

I can think of two things that might be improved here:

1. Don't let FakeQuerySets leak: I don't get them on the console and they're not documented as far as I can see, so why do I get one? The issue is repeatable for me, so I can investigate if needed.
2. A FakeQuerySet might want to be able to delete its contents.

**Comment by johannes-torchbox** [comment]
One more note: I'm using `getattr` to retrieve the related items, effectively this:

    getattr(model_name, 'related_name').all().delete()

Do you maybe perform getattr-magic?

**Comment by gasman** [comment]
Hi Johannes,
You'll only receive a FakeQuerySet if you've previously written to the relation (e.g. with `model_name.related_name.add(...)`) and not saved it. If the model object matches the database state, you'll get a real queryset (see https://github.com/torchbox/django-modelcluster/blob/master/modelcluster/fields.py#L52-55 - `_cluster_related_objects` is populated when you write to the relation).

"A FakeQuerySet might want to be able to delete its contents." - yes, ideally FakeQuerySet would implement the full QuerySet API (except where it absolutely requires a SQL backend, such as `extra`), but it currently only implements the most common read operations. In this case, though, I'm not sure if there's a legitimate reason to write to a relation and then call `delete`, or whether it will only ever be done in error...

**Comment by johannes-torchbox** [comment]
Ah, ok, I understand. I solved my problem in the meantime with another solution, though. =)

The situation I had when encountering this was that I saved my original object through a ModelForm (with `commit=False`) and then worked on the related sets. However, afaics, the ModelForm touches the related objects, which is why I got a FakeQuerySet in the first place. Probably not enough of a use case, since calling `.delete` effectively is the same as ` = []`.
